packages/server/api/src/app/tables/field/field.controller.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    ALL_PRINCIPAL_TYPES,
+    CreateFieldRequest,
+} from '@activepieces/shared'
+import { FastifyPluginAsyncTypebox, Type } from '@fastify/type-provider-typebox'
+import { StatusCodes } from 'http-status-codes'
+
+export const fieldController: FastifyPluginAsyncTypebox = async (fastify) => {
+
+    fastify.post('/:id/fields', CreateRequest, async (request, reply) => {
+        await reply.status(StatusCodes.OK).send({})
+    },
+    )
+
+    fastify.get('/:id/fields/:fieldId', GetFieldByIdRequest, async (request, reply) => {
+        await reply.status(StatusCodes.OK).send({})
+    },
+    )
+
+    fastify.delete('/:id/fields/:fieldId', DeleteFieldRequest, async (request, reply) => {
+        await reply.status(StatusCodes.OK).send({})
+    },
+    )
+
+    fastify.get('/:id/fields', GetFieldsRequest, async (request, reply) => {
+        await reply.status(StatusCodes.OK).send({})
+    },
+    )
+}
+
+const CreateRequest = {
+    config: {
+        allowedPrincipals: ALL_PRINCIPAL_TYPES,
```

Comment:
Let's only Allow [ENGINE, USER]

The unknown means unauthenticated and api key is not applicable.